### ISSUE TITLE
Sparse / Partial Masters and Component Glyphs

### ISSUE THREAD

**Issue by weiweihuanghuang** [googlefonts/fontmake#606]
When I have a sparse master but the glyph is only components (the position of the components need to be edited and same with an anchor), does the glyph that it references need to be in the master too?

**Comment by weiweihuanghuang** [comment]
Hmm gonna assume that the answer is yes because I no longer have the problems to do with compatibility...

**Comment by anthrotype** [comment]
Yes, otherwise the component references a missing base glyph. Component references in UFO only work within the same layer.

If you only want to variate the components offsets within a composite glyph, but not the outlines of the base glyphs, then you need to include the base glyphs in the sparse master unchanged. Deltas will be added only for the component offsets.

**Comment by anthrotype** [comment]
> you need to include the base glyphs in the sparse master unchanged. Deltas will be added only for the component offsets.

Hmm maybe I'm wrong.. I need to think about this a bit more. You raise a valid point.
